### PR TITLE
docs(cli): added makefile and go file to create docs

### LINKED ISSUE #3874: automate cli docs generation
followup from https://github.com/aquasecurity/trivy/pull/3815, we would need to automate the creation of documentation for cli commands

```go
err := doc.GenMarkdownTree(rootCmd, "./docs/docs/references/cli")
	if err != nil {
		log.Fatal(err)
	}
```

> I think the best approach would be to create a simple go file which will be go run from the Makefile.

### PR DESCRIPTION
**Pull request by krishnaduttPanchagnula**
## Description

Automated the creation of documentation using Makefile. This is in relation to the issue [#3874](https://github.com/aquasecurity/trivy/issues/3874)

## Related issues
- Close #3874

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.

### CODE REVIEW

**Comment by knqyf263** [comment]
Thanks for your contribution! We were preparing for this document generation.
https://github.com/aquasecurity/trivy/pull/3932

After my change, you can write Go code in magefile. Just a moment, please.

**Comment by knqyf263** [comment]
Now, it's in the main branch. Could you move the logic here?
https://github.com/aquasecurity/trivy/blob/b40f60c4058ce88e0a3ed3a16196d8c60a5b1a88/magefiles/magefile.go#L357-L361

**Comment by krishnaduttPanchagnula** [comment]
Will incorporate changes done by @knqyf263. Closing this current PR to raise a new one.

**Comment by knqyf263** [comment]
You can keep using this branch and PR.

**Comment by krishnaduttPanchagnula** [comment]
I have one question, should i also add changes to go.sum and go mod file to Pr ?

File: magefiles/magefile.go [hunk 1]
```diff
@@ -356,7 +356,11 @@ func (Docs) Serve() error {
 
 // Generate generates CLI references
 func (Docs) Generate() error {
-	// TODO
+	
+	if err := sh.Run("go", "run", "docs/doc_creator/main.go"); err != nil {
+		  
+		return err
+	}
```
Comment (knqyf263) [inline magefiles/magefile.go:363]:
```suggestion
	err := doc.GenMarkdownTree(commands.NewApp("v0.38"), "./docs/docs/references/cli")
	if err != nil {
		return err
	}
```

And the version is fixed to v0.38. We should take it from git tag.

**Comment by knqyf263** [comment]
>I have one question, should i also add changes to go.sum and go mod file to Pr ?

Is there any reason to update go.mod?

**Comment by krishnaduttPanchagnula** [comment]
> Is there any reason to update go.mod?

I am adding one indirect dependency go-md2man. Also added version as per your suggestion via function. Please review and suggest if any changes are to be made.

File: go.mod [hunk 2]
```diff
@@ -199,6 +199,7 @@ require (
 	github.com/containerd/stargz-snapshotter/estargz v0.14.3 // indirect
 	github.com/containerd/ttrpc v1.1.0 // indirect
 	github.com/containerd/typeurl v1.0.2 // indirect
+	github.com/cpuguy83/go-md2man/v2 v2.0.2 // indirect
```
Comment (knqyf263) [inline go.mod:202]: You also need to run `go mod tidy`.

File: magefiles/magefile.go [hunk 3]
```diff
@@ -356,7 +358,14 @@ func (Docs) Serve() error {
...
 func (Docs) Generate() error {
-	// TODO
+	ver, errv := version()
+	if errv != nil {
+		return errv
+	}
+	err := doc.GenMarkdownTree(commands.NewApp(ver), "./docs/docs/references/cli")
+	if err != nil {
+		return err
+	}
```
Comment (knqyf263) [inline magefiles/magefile.go:368]: I found a flag to disable auto gen tag.

```suggestion
	ver, err := version()
	if err != nil {
		return err
	}
	cmd := commands.NewApp(ver)
	cmd.DisableAutoGenTag = true
	if err = doc.GenMarkdownTree(cmd, "./docs/docs/references/cli"); err != nil {
		return err
	}
	return nil
```

File: go.mod (hunk 2 above)
Comment (knqyf263) [reply go.mod:202]: go.sum will be updated

Comment (krishnaduttPanchagnula) [reply go.mod:202]: Yes, i pushed the updated go.sum file too

Comment (knqyf263) [reply go.mod:202]: Thanks

File: magefiles/magefile.go [hunk 4]
```diff
@@ -9,9 +9,11 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/aquasecurity/trivy/pkg/commands"
```
Comment (knqyf263) [inline magefiles/magefile.go:12]: Imports must be grouped as below

```
std packages

3rd-party packages

trivy packages
```

Comment (krishnaduttPanchagnula) [reply magefiles/magefile.go]: updated it.

**Review by knqyf263 — APPROVED** [review]
LGTM👍

**Pull request merged**